lib/Infer/DataflowPruning.cpp

```diff
@@ -0,0 +1,330 @@
+#include "souper/Infer/DataflowPruning.h"
+
+namespace souper {
+namespace {
+using souper::dataflow::EvalValue;
+using souper::dataflow::ValueCache;
+
+EvalValue EvaluateInst(souper::Inst* Inst, std::vector<EvalValue> args) {
+  switch (Inst->K) {
+    case souper::Inst::Const:
+      return {Inst->Val, false, false};
```

Comment:
when passing a boolean flag, include a comment saying what it's about

Reply:
Switched to llvm::Optional, no more magic flags!